format/view-spec.md

```diff
@@ -42,12 +42,28 @@ An atomic swap of one view metadata file for another provides the basis for maki
 
 Writers create view metadata files optimistically, assuming that the current metadata location will not be changed before the writer's commit. Once a writer has created an update, it commits by swapping the view's metadata file pointer from the base location to the new location.
 
+### Materialized Views
+
+Materialized views are a type of view with precomputed results from the view query stored as a table.
+When queried, engines may return the precomputed data for the materialized views, shifting the cost of query execution to the precomputation step.
+
+Iceberg materialized views are implemented as a combination of an Iceberg view and an underlying Iceberg table, known as the storage table, which stores the precomputed data.
+The metadata for a materialized view extends the Iceberg view metadata, adding a pointer to the precomputed data and refresh information to determine if the data is still fresh. 
+The refresh information is composed of data about the so-called "source tables", which are the tables referenced in the query definition of the materialized view. 
+The storage table can be in the states of "fresh", "stale" or "invalid", which are determined from the following situations:
+* **fresh** -- The `snapshot_id`s of the last refresh operation match the current `snapshot_id`s of the source tables.
+* **stale** -- The `snapshot_id`s do not match, indicating that a refresh operation needs to be performed to capture the latest source table changes.
```

Comment:
Should we consider the case where the current materialized view references another materialized view? If we determine that the current materialized view is not stale, we should also ensure that any referenced materialized views are not marked as invalid.

Reply:
it seems that from the rest of the spec, nested view is indeed a valid use case, for materialized view referring other materialized views or even common views. However it seems to me that the rest of the spec didn't mention one of these states being recorded anywhere in the spec, which I would also agree to not include since it feels very challenging to ensure this info could be trustworthy especially when there are multiple engines doing reads/writes on the same data. With this, do we mention these `fresh`/`stale`/`invalid` here mostly for establishing a nomenclature for view users? If so I wonder if it's worth explicitly mentioning that to avoid the user expecting one of these three states to be recorded in their view metadata.

Reply:
Thanks @chenjian2664 and @yyanyy for your comments. These are great points. The current definition supports nested views in that it fully expands the query tree at refresh time and stores all directly and indirectly referenced tables in the refresh-state. This can include multiple levels of views.
This currently would also apply to materialized views. However, I think you're making a good point. When referencing another materialized view it would be easier to just store the snapshot_id of its storage table instead of tracking all table references in the materialized view.

You're correct in assuming that `fresh`/`stale`/`invalid` are introduced to have some nomenclature, these to not occur in the metadata. I will try to make this clearer.

Reply:
Which option do you prefer? And should I include this in the spec?

Reply:
Personally, I prefer the 2nd option (treating source MV as a table). 

But we need to wait for more people to weigh in before we make any change to this PR.